### PR TITLE
Fixed handling re-issued credentials

### PR DESCRIPTION
**Pull request by lukachi**
VC clear methods:
save vc & clear vc methods have similar ways to build query hash for ceramic compose db.
If VCs with specific queryHash had been found - we return them instead of generating new ones.

But in cases, when the issuer has migrations, or some other `VC's changing situations - it could still return the same query hash, but other necessary data could be outdated.

To deal with that, specific checks had been added to remove all matched VC's from ceramic, and save new ones

### CODE REVIEW

**Comment by ihordiachenko** [comment]
@lukachi add a proper PR name & description

File: packages/snap/src/helpers/credential-helpers.ts [hunk 1]
```diff
@@ -454,3 +508,13 @@ export const loadDataByUrl = async (
 
   return await response.json();
 };
+
+export const isVCsV2 = (vcs: W3CCredential[]) => {
+  return vcs.every((vc) => {
+    return (
+      vc.issuer.includes('readonly') &&
+      // FIXME: temp hotfix
+      new Date(vc.issuanceDate!).getFullYear() > 2023
```
Comment (ihordiachenko) [inline packages/snap/src/helpers/credential-helpers.ts]: Add a detailed comment that explains the reasoning behind this

File: packages/snap/src/helpers/credential-helpers.ts [hunk 2]
```diff
@@ -245,6 +248,57 @@ export class VCManager {
...
+      this.ceramicProvider,
+    );
+
+    const ids = data
+      .map((el) => el.verifiableCredentialIndex?.edges)
+      .flat()
+      .map((el) => el?.node?.id ?? '')
+      .filter((el) => Boolean(el));
+
+    await Promise.all(
```
Comment (ihordiachenko) [inline packages/snap/src/helpers/credential-helpers.ts:317]: Do you just delete all credentials matching the query? I'd rename the method if that's true

File: packages/connector/src/types.ts [hunk 3]
```diff
@@ -87,7 +87,7 @@ export type ProofQuery = {
   claimId?: string;
   credentialSubjectId?: string;
   context?: string;
-  type?: string;
+  type?: string[];
```
Comment (ihordiachenko) [inline packages/connector/src/types.ts:90]: Can iden3 really handle multiple types right now?

File: packages/snap/src/helpers/credential-helpers.ts (hunk 2 above)
Comment (ihordiachenko) [reply packages/snap/src/helpers/credential-helpers.ts:317]: @lukachi

File: packages/connector/src/types.ts (hunk 3 above)
Comment (lukachi) [reply packages/connector/src/types.ts:90]: it's the way issuer build type property in VC

**Review by ihordiachenko — APPROVED** [review]

**Pull request merged**